applogic/src/api/mobile_logging.rs

```diff
@@ -79,14 +79,13 @@ pub struct SendToDartLogger {
 
 impl SendToDartLogger {
     pub fn set_stream_sink(stream_sink: StreamSink<LogEntry>) {
-        let mut guard = SEND_TO_DART_LOGGER_STREAM_SINK.write().unwrap();
-        let overriding = guard.is_some();
-
-        *guard = Some(stream_sink);
-
-        drop(guard);
-
-        if overriding {
+        let prev_stream_sink = {
+            let mut guard = SEND_TO_DART_LOGGER_STREAM_SINK.write().expect("poisoned");
+            // Note: previous stream sink MUST NOT be dropped before the `guard` is released.
+            // On drop, it will log and because the `RwLock` is not reentrant, this will deadlock.
```

Comment:
```suggestion
            // On drop, it will lock and because the `RwLock` is not reentrant, this will deadlock.
```

Reply:
Actually, I really mean `it will log a message`. I will adjust the comment.